gcc/rust/gcc-rust/build.rs

```diff
@@ -26,10 +26,12 @@ fn main() {
         .whitelist_function("make_signed_type")
         .whitelist_function("make_unsigned_type")
         .whitelist_function("get_identifier")
+        .blacklist_type("poly_int64")
         .default_enum_style(Rust {
             non_exhaustive: false,
         })
         .raw_line("#![allow(dead_code, non_upper_case_globals, non_camel_case_types)]")
+        .raw_line("pub type poly_int64 = i64;")
```

Comment:
Won't this cause an i64 to be passed over the FFI boundary instead of a poly_int64 instance? The poly_int64 constructor needs to be called. (I think the constructor call will also fail to compile when HOST_WIDE_INT is smaller than i64)

Reply:
The docs said a poly_int<1,T> is just a T: 
https://gcc.gnu.org/onlinedocs/gccint/Consequences-of-using-poly_005fint.html#Consequences-of-using-poly_005fint

HOST_WIDE_INT is defined as long long for host that have it and long otherwise, so this is equivalent to an isize right ?

Reply:
> HOST_WIDE_INT is defined as long long for host that have it and long otherwise, so this is equivalent to an isize right ?

Actually, I just found this in `hwint.h`:
```c
/* Set HOST_WIDE_INT, this should be always 64 bits.
   The underlying type is matched to that of int64_t and assumed
   to be either long or long long.  */
```

So it's equivalent to an i64 :)

> The docs said a poly_int<1,T> is just a T

I think maybe they meant that at code generation time it's treated as T. It's implemented as a `template class poly_int` that inherits from `template struct poly_int_pod` that contains the coefficient array. And I don't see any virtual methods. So I guess it just wraps a single T. And probably the ABI is the same, but it still seems unsafe to me. I think maybe it would be better to pass an i64 to a C++ wrapper function and have the C++ code do the cast to poly_int (either implicitly or explicitly) before calling `build_int_cst_type`.

Reply:
I added the wrapper, this is good for merging or do you have any others suggestions ?